internal/scanner/scanner.go

```diff
@@ -290,46 +298,99 @@ func (s *Scanner) ScanClusterWideResources(ctx context.Context, runUID string) e
 	return nil
 }
 
-func (s *Scanner) auditResource(ctx context.Context, policies []*policies.Policy, resource unstructured.Unstructured, runUID string) {
+type policyAuditResult struct {
+	policyReport    *wgpolicy.PolicyReport
+	policy          policiesv1.Policy
+	admissionReview *admissionv1.AdmissionReview
+	errored         bool
+}
+
+//nolint:funlen
+func (s *Scanner) auditResource(ctx context.Context, policies []*policies.Policy, resource unstructured.Unstructured, runUID string) error {
+	log.Info().
+		Str("resource", resource.GetName()).
+		Dict("dict", zerolog.Dict().
+			Int("policies to evaluate", len(policies)).
+			Int("parallel policies audit", s.parallelPoliciesAudits),
+		).Msg("audit resource")
+
 	policyReport := report.NewPolicyReport(runUID, resource)
+	semaphore := semaphore.NewWeighted(int64(s.parallelPoliciesAudits))
+	var workers sync.WaitGroup
 
-	for _, p := range policies {
-		url := p.PolicyServer
-		policy := p.Policy
+	auditResults := make(chan policyAuditResult)
+	reportsDone := make(chan bool)
+
+	go func() {
```

Comment:
do we need this go routine here?
couldn't we just 
```go
for res := range auditResults {
		report.AddResultToPolicyReport(res.policyReport, res.policy, res.admissionReview, res.errored)
	}
```

after the workers have completed the auditing?

Reply:
Done